src/runtime_src/core/tools/common/Report.cpp

```diff
@@ -99,7 +100,7 @@ Report::getFormattedReport( const xrt_core::device *pDevice,
       std::cerr << reportName << std::endl;
     }
 
-    std::cerr << "  ERROR: " << e.what() << std::endl << std::endl;
-    // Fall through
+    std::cerr << "  ERROR: " << e.what() << std::endl;
+    throw xrt_core::error(std::errc::operation_canceled);

```

Comment:
Hmm....
With this change, if any report throughs an exception, then any reports that follow it will not be performed.  This is unfortunately not the expected use case for ALL reports should be executed.

Instead, could you  refactor the code to allow for ALL of the reports to be ran and if "any of them" produces an exception, then throw the `xrt_core::error(std::errc::operation_canceled)` exception.

Reply:
Ok makes sense, I had a feeling that was the case but was not sure.